client/generator-carousel-admin/js/main.js

```diff
@@ -4,6 +4,44 @@ var default_url      = 'https://en.wikipedia.org/wiki/Financial_Times';
 var default_duration = 10;
 var default_url = "https://en.wikipedia.org/wiki/Static_web_page";
 
+var keyUpTimeout = null,
+	table = undefined;
+
+function appendNewInputToForm(where){
+
+	var existingInput = table.getElementsByClassName('url-and-duration')[1],
+		clone = existingInput.cloneNode(true),
+		inputs = clone.getElementsByTagName('input');
+
+	for(var g = 0; g < inputs.length; g += 1){
+		inputs[g].value = "";
+	}
+
+	table.appendChild(clone);
+
+}
+
+function allInputsHaveContent(inputs){
```

Comment:
I reckon with this new functionality, we could strip back the initial size of the form to a single input and let it grow with however many the user requires.